lib/puppet/provider/pulp_rpmrepo/api.rb

```diff
@@ -1,10 +1,18 @@
-require File.expand_path('../../../util/pulp_util', __FILE__)
+require File.expand_path('../../../util/repo_provider', __FILE__)
 
-Puppet::Type.type(:pulp_rpmrepo).provide(:api) do
+Puppet::Type.type(:pulp_rpmrepo).provide(:api, :parent => PuppetX::Pulp::RepoProvider) do
   commands :pulp_admin => '/usr/bin/pulp-admin'
 
   mk_resource_methods
 
+  def self.repo_type
```

Comment:
I think that this is very ugly, but I don't know enough about ruby inheritance to do this cleaner.